### PR TITLE
Handle Terminate messages appropriately

### PR DESCRIPTION
**Pull request by elprans**
The `Terminate` protocol message (`b'X'`) is the "negotiated" connection 
termination, when the client asks the server to disconnect and waits for 
the connection to drop.  This ensures that both the client and the server
are in agreement with respect to whether a connection between them still
exists, which is important for cases when a command such as
`DROP DATABASE` immediately follows the client disconnect.

Depends on edgedb/edgedb-python#166

Fixes: #2246

### CODE REVIEW

File: edb/server/protocol/binary.pyx [hunk 1]
```diff
@@ -298,7 +298,10 @@ cdef class EdgeConnection:
             self._transport.abort()
             self._transport = None
 
-    async def close(self):
+    def close(self):
+        if self.dbview is not None:
```
Comment (1st1) [inline edb/server/protocol/binary.pyx:302]: Hm, we need to put this in `connection_lost`.

Comment (elprans) [reply edb/server/protocol/binary.pyx:314]: Done. I've added the necessary bits to `abort()` and made sure it always gets called.  The `pinned_pgcon` abort business has been moved there as well.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -243,7 +243,7 @@ jobs:
 
     - name: Install edgedb-server and populate egg-info
       env:
-        CACHE_HIT: ${{ steps.postgres-cache.outputs.cache-hit }}
+        CACHE_HIT: ${{ steps.venv-cache.outputs.cache-hit }}
```
Comment (1st1) [inline .github/workflows/tests.yml:246]: Let's move into a separate commit

**Review by 1st1 — APPROVED** [review]
